### PR TITLE
Initial merge to ONS repo

### PR DESCRIPTION
**Pull request by KamenDimitrov97**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+
+#### Download or create embeddings
+
+These are most simply sourced as [pretrained fifu models](https://finalfusion.github.io/pretrained), but can be dynamically generated
+using the embedded FinalFusion libraries.
+
+To build and run the API using docker:
+
+```
+make run
```
Comment (lindenmckenzie) [inline README.md:21]: there is no `make run` command in the makefile.

**Review by lindenmckenzie — COMMENTED** [review]
Some general comments prior to full review.

**Review by philtweir — CHANGES_REQUESTED** [review]
I'm not sure if you can see this before I submit it so, more to come, but keen you can begin working through!

File: Dockerfile.in [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+## STAGE 1 - Core package(s)
+
+FROM flaxandteal/ff_fasttext_build:latest as ff
+
+RUN maturin build
+
+## STAGE 2 - API
+
+FROM python:3.9
```
[the remark this answers is not in the document]
Comment (lindenmckenzie) [reply Dockerfile.in:12]: I don't believe we have a standard so makes sense to go latest LTS version as you say.

File: ff_fasttext_api/healthcheck.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+            'version': self.version,
+            'uptime': self.get_uptime(),
+            'start_time': self.start_time,
+            'checks': self.checks
+        }
+
+        return response
+    
+    def get_last_commit(self):
+        last_commit = subprocess.check_output(['git', 'rev-parse', 'HEAD'])
```
[the remark this answers is not in the document]
Comment (lindenmckenzie) [reply ff_fasttext_api/healthcheck.py:36]: https://github.com/ONSdigital/dp-topic-api/blob/1eee0e9eb4cdd9ab5898eb1051d51095851812eb/Makefile#LL10C1-L11C1 <-- is done int he makefile.

File: translate_cache.py
Comment (philtweir) [inline translate_cache.py:1]: This should live somewhere better, probably bonn-py`

File: tests/api/server_test.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
+import pytest
+from fastapi.testclient import TestClient
+from ff_fasttext_api.server import make_app
+from unittest.mock import MagicMock
+from bonn.extract import CategoryManager
+from ff_fasttext_api.healthcheck import Healthcheck
+
+@pytest.fixture(scope='module')
+def test_client():
+    # Create a mock CategoryManager object to simulate the FastText model
```
Comment (philtweir) [inline tests/api/server_test.py:10]: This is useful, but an integration test that uses the sample data with the real category manager (bonn) would be good -- for instance, there is no test guaranteeing that our assumed return structure is what bonn would send (interface mismatch is a common pitfall for an overly encapsulated test suite, but equally, we can't depend on the entire environment - hence, a minimal sample fifu model is in the code base)

**Pull request merged**